Library/Homebrew/tap.rb

```diff
@@ -1352,7 +1352,12 @@ def core_cask_tap?
 
   sig { params(token: String).returns(Pathname) }
   def new_cask_path(token)
-    cask_subdir = token[0].to_s
+    cask_subdir = if token.start_with?("font-")
+      "fonts/#{token.delete_prefix("font-").first}"
```

Comment:
```suggestion
    cask_subdir = if token.start_with?("font-")
      "font-/#{token.delete_prefix("font-").first}"
```

Alternatively:
```suggestion
    cask_subdir = if token.start_with?("font")
      "font/#{token.sub(/^font-?/).first}"
```

or:
```suggestion
    cask_subdir = if token.start_with?("font-")
      "font-#{token[5]}"
```

I think the directory name should always match the prefix.